src/Sulu/Bundle/MediaBundle/Controller/CollectionController.php

```diff
@@ -77,13 +77,26 @@ public function getAction($id, Request $request)
     {
         try {
             $locale = $this->getLocale($request);
-            $depth = $request->get('depth');
+            $depth = intval($request->get('depth', 0));
             $breadcrumb = $this->getBooleanRequestParameter($request, 'breadcrumb', false, false);
             $collectionManager = $this->getCollectionManager();
+
+            // filter children
+            $listRestHelper = $this->get('sulu_core.list_rest_helper');
+            $limit = intval($request->get('limit', $listRestHelper->getLimit()));
+            $offset = intval(($request->get('page', 1) - 1)) * $limit;
+            $search = $request->get('search', '');
```

Comment:
`$listRestHelper->getSearchPattern()` would do quite the same, except it delivers `null` if the param is not given, which would also be cleaner IMHO